lib/dotcom/trip_plan/alerts.ex

```diff
@@ -35,33 +47,66 @@ defmodule Dotcom.TripPlan.Alerts do
     |> Enum.uniq()
   end
 
-  defp leg_entities(%Leg{mode: mode} = leg) do
+  def leg_entities(%Leg{mode: mode} = leg) do
     for entity <- mode_entities(mode),
         stop_id <- Leg.stop_ids(leg) do
       %{entity | stop: stop_id}
     end
   end
 
-  defp mode_entities(%TransitDetail{route: route, trip: %{id: trip_id}}) do
-    trip =
-      if is_nil(route.external_agency_name) do
-        Schedules.Repo.trip(trip_id)
-      end
+  defp leg_entities_from(%Leg{mode: mode} = leg) do
+    for entity <- mode_entities(mode),
+        stop_id <- Leg.stop_ids_from(leg) do
+      %{entity | stop: stop_id}
+    end
+  end
 
-    route_type =
-      if route do
-        route.type
-      end
+  defp leg_entities_to(%Leg{mode: mode} = leg) do
+    for entity <- mode_entities(mode),
+        stop_id <- Leg.stop_ids_to(leg) do
+      %{entity | stop: stop_id}
+    end
+  end
 
-    direction_id =
-      if trip do
-        trip.direction_id
-      end
+  defp mode_entities(%TransitDetail{route: route}) when is_external?(route) do
+    []
+  end
 
-    [%IE{route_type: route_type, route: route.id, trip: trip_id, direction_id: direction_id}]
+  defp mode_entities(%TransitDetail{
+         route: %{id: route_id, type: route_type},
+         trip: %{id: trip_id, direction_id: direction_id}
+       }) do
+    [%IE{route_type: route_type, route: route_id, trip: trip_id, direction_id: direction_id}]
   end
 
   defp mode_entities(_) do
     []
   end
+
+  def by_mode_and_stops(alerts, leg) do
+    grouped_alerts =
+      alerts
+      |> Enum.group_by(fn alert ->
+        alert.informed_entity.entities
+        |> Enum.any?(fn
+          %{stop: nil} -> false
+          _ -> true
+        end)
+      end)
+
+    route_alerts = grouped_alerts[false] || []
+    stop_alerts = grouped_alerts[true] || []
```

Comment:
[Done!](https://github.com/mbta/dotcom/pull/2269/commits/61c70b5922069e3b56c76e0580c92c32e1dfdb5e) I also thought it was clearer to reverse the sense of the check, and use `Enum.all?` instead of `Enum.any?` (not sure